src/client/js/partials/add-email.js

```diff
@@ -0,0 +1,53 @@
+let dialogEl, form
+
+function init () {
+  dialogEl = document.querySelector('dialog[data-partial="add-email"]')
+  if (!dialogEl) return
+
+  form = dialogEl.querySelector('form')
+  form.addEventListener('submit', handleSubmit)
+  dialogEl.addEventListener('close', kill)
+}
+
+async function handleSubmit (e) {
+  e.preventDefault()
+
+  try {
+    form.elements['email-submit'].toggleAttribute('disabled', true)
+    const res = await fetch('/api/v1/user/email', {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-csrf-token': form.elements['csrf-token'].value
+      },
+      mode: 'same-origin',
+      method: 'POST',
+      body: JSON.stringify({
+        email: form.elements['email-address'].value
+      })
+    })
+
+    if (!res.ok) throw new Error('Bad fetch response')
```

Comment:
Question (non-blocking): Should we move and consolidate new errors in `src/utils/error.js`?

Reply:
Yes, I might need a ticket and a tech discussion on this to align – I out of the loop on the patterns we've established!